### ISSUE TITLE
Bug: Can't train with NVIDIA GPU Ampere Architecture

### ISSUE THREAD

**Issue by dalberpi** [coqui-ai/STT#2257]
Hi all,

I’m facing an issue, when I train, only the CPUs are used and not my GPU. I have followed rigorously the playbook to set up the training environment for using GPU. So the reason why it’s not using GPU, is because its architecture is not compatible with cuda 10 which is in the release you provide. 
So, I’m trying to build the Dockerfile to have a cuda 11.x version in the container and to be able to train using my GPU. I succeeded to build the Dockerfile with the cuda 11.3 version so in my container when I do : `nvcc --version` it displays that there is cuda 11.3 version. Nevertheless, when I train, the GPU is still not used.

This is my Dockerfile I've built :
[Dockerfile.txt](https://github.com/coqui-ai/STT/files/9062397/Dockerfile.txt)

**Expected behavior**
I would like to train with my RTX 3090 GPU (Ampere Architecture) but it’s not compatible with cuda 10 so when I train it only uses CPU.

**Environment (please complete the following information):**
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 18.04
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: upstream TensorFlow
- **TensorFlow version (use command below)**: 2.6.2
- **Python version**: 3.6.9
- **Bazel version (if compiling from source)**: 4.2.1
- **GCC/Compiler version (if compiling from source)**: 7.5.0
- **CUDA/cuDNN version**: CUDA-11.7/cuDNN-8.4.1
- **GPU model and memory**: NVIDIA GeForce RTX 3090, 24 GB

**Comment by dalberpi** [comment]
Unfortunately when I use this docker image `ghcr.io/coqui-ai/stt-train:v0.10.0-alpha.4` it's not working. When I create the docker container from this image by using `--gpus all` flag : when I want to train, my GPU memory is saturated and it doesn't train but when I'm not using this flag it's only running with CPUs. Do you know from you've heard for the people it worked if they have done something special ?

**Comment by wasertech** [comment]
Also you may need to allow memory growth on this one.

```
use_allow_growth: bool = field(
        default=False,
        metadata=dict(
            help="use Allow Growth flag which will allocate only required amount of GPU memory and prevent full allocation of available GPU memory"
        ),
    )
```

See https://stt.readthedocs.io/en/latest/TRAINING_FLAGS.html?highlight=Allow%20growth%20#flags

**Comment by dalberpi** [comment]
Oh yes, I didn't notice that it was a very old image...
So, as you adviced me, I downloaded `docker pull ghcr.io/coqui-ai/stt-train:main` and when I do : `nvcc --version` in the docker container created from this image I have 11.6 version. Then, I've tried to train especially with the flags `--use_allow_growth`, `--train_batch_size`, `--dev_batch_size`. And now the training with GPU usage is working ! So the training should finish properly.
Thank you very much for your help !
